File: factorial.js
// const prompt = require('prompt-sync')();
const input = Number(prompt('enter the num: '));
function factorial(input) {
  try {
    console.log(input);
    if (typeof input !== 'number') throw Error('input must a number');
    if (input <= 1) {
      return 1;
    } else {
      return input * factorial(input - 1);
    }
  } catch (error) {
    console.error(error);
  }
}
console.log(factorial(input));

File: factorial.py


input = input("enter the number: ")
input_converted = int(input)


def factorial(num):
    # execute this block code first
    try:
        if (num <= 1):
            return 1
        else:
            return num * (factorial(num - 1))
    # if error's occured this block code will execute
    except:
        return "there is an error"


output = factorial(input_converted)
# print some output
print(output)
